sdk/log/record.go

```diff
@@ -48,6 +63,10 @@ type Record struct {
 	//   - Unused array elements are zero-ed. Used to detect mistakes.
 	back []log.KeyValue
 
+	// dropped is the count of attributes that have been dropped when limits
+	// were reached.
+	dropped int
```

Comment:
Adding "limiting" and "deduplication" will also affect the unit testing testing experience of exporters and custom processors mentioned in https://github.com/open-telemetry/opentelemetry-go/issues/5192.

_I am starting to doubt if it is really bad that testing processors and exporters requires creating a logger provider._

Reply:
Is this just a general comment, or are you suggesting we change something here?

Reply:
General comment. Nothing to change in this PR. It would be good to update the description of https://github.com/open-telemetry/opentelemetry-go/issues/5192 after this is merged.